server/workers/shareScrollWorker.ts

```diff
@@ -0,0 +1,548 @@
+const BANNER_HEIGHT = 50;
+const FRAME_DELAY = 100;
+const SCROLL_STEP = 2;
+
+async function fileExists(filePath: string) {
+  try {
+    await fs.stat(filePath);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+parentPort?.on('message', async function (message) {
+  if (message.type !== 'banner') return;
+  const { username, filePath, token } = message;
+  try {
+    await generateBannerToTemporary(username, filePath, token);
+    await moveBannerFromTemporary(filePath);
+    parentPort?.postMessage({ type: 'success', username });
+  } catch (e) {
+    parentPort?.postMessage({ type: 'error', username, error: e });
+  } finally {
+    parentPort?.postMessage({ type: 'jobFinished', username });
+  }
+});
+
+async function moveBannerFromTemporary(filePath: string) {
+  try {
+    if (await fileExists(filePath)) {
+      await fs.unlink(filePath);
+    }
+    await fs.rename(`${filePath}.tmp`, filePath);
+  } catch (error) {
+    console.error('Error moving banner from temporary:', error);
+    throw error;
+  }
+}
+
+// utils
+
+function createEmptyFrame(width: number, height: number): sharp.Sharp {
+  return sharp({
+    create: {
+      width,
+      height,
+      channels: 4,
+      background: { r: 0, g: 0, b: 0, alpha: 0 },
+    },
+  });
+}
+
+async function textToPng(
+  text: string,
+  font: string, // eg. "16px Alkhemikal"
+  styles: string // eg: "fill: white; ..."
+): Promise<Buffer> {
+  const { width, height } = await sharp(
+    Buffer.from(
+      `<svg>
+        <text style="font: ${font};">${text}</text>
+      </svg>`
+    )
+  )
+    .png()
+    .metadata();
+  // make a dummy png with the text,
+  // the text renders ABOVE the viewport which isn't what we want,
+  // (and sharp doesn't support the dominant-baseline property!)
+  // and there's no way for a svg to grab its own children's dimensions,
+  // so the purpose of this dummy is to provide the raw dimensions of the text.
+
+  const pngBuffer = await sharp(
+    Buffer.from(
+      `<svg width="${(width ?? 0) + 1}" height="${(height ?? 0) + 4}">
+        <style>
+          .text {
+            font: ${font};
+            filter: drop-shadow(1px 1px 0px rgba(0, 0, 0, 0.25));
+            ${styles}
+          }
+        </style>
+        <text y="${(height ?? 0) + 1}" class="text">${text}</text>
+      </svg>`
+    )
+  )
+    .png()
+    .toBuffer();
+  // the real deal is here. it uses the dummy-given dimensions
+  // to push down the text into the viewport at an appropriate distance
+  // and adds spacing to accomodate the text-shadow.
+
+  return pngBuffer;
+}
+
+// the meat of it
+
+async function generateBannerToTemporary(
+  username: string,
+  filePath: string,
+  token: string
+) {
+  // missing things currently:
+  // - complete the banner generation chain, one func at a time
+  // - get access token from user so we can count a hidden scroll's dragons
+  // - how to hit the hatchery's api for click info? flair info and images?
```

Comment:
I think the easiest approach will be passing it down to the worker instead of trying to import the db in the worker thread. that also then gives you the opportunity to validate the user is actually registered with the garden.